addons/stock/tests/test_quant.py

```diff
@@ -70,6 +70,19 @@ def gather_relevant(self, product_id, location_id, lot_id=None, package_id=None,
         quants = self.env['stock.quant']._gather(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
         return quants.filtered(lambda q: not (q.quantity == 0 and q.reserved_quantity == 0))
 
+    def test_copy_quant(self):
+        """
+        You should not be allowed to duplicate quants.
+        """
+        warehouse = self.env.ref('stock.warehouse0')
+        quant = self.env['stock.quant'].create([{
+            'location_id': warehouse.lot_stock_id.id,
```

Comment:
Ah sorry, just noticed this. It's safer to not refer to a demo/data record in tests in case they are changed in the future or don't exist when the test is run (e.g. test without demo data) if possible. May as well use the location created specifically for this batch of tests in the setup.

```suggestion
        quant = self.env['stock.quant'].create([{
            'location_id': self.stock_location.id,,
```

Reply:
@ticodoo Good point. Applied :D